flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/NettyShuffleMaster.java

```diff
@@ -45,6 +56,9 @@ public class NettyShuffleMaster implements ShuffleMaster<NettyShuffleDescriptor>
 
     private final int networkBufferSize;
 
+    private final Map<IntermediateDataSetID, Collection<NettyShuffleDescriptor>>
```

Comment:
Previously for TaskExecutor managed partitions, the process of prompting should be implemented in `JobMasterPartitionTracker` directly, thus the `NettyShuffleMaster` should not need to manage the cluster partitions directly? 

Also I'm a bit wondering if it is possible we unify the two processes?

Reply:
If I understand correctly, the life cycle of the `JobMasterPartitionTracker` is bound to a job. It cannot keep the mapping from `IntermediateDataSetID` to `ShuffleDescriptors` of the cluster partitions across jobs.